src/mam4xx/nucleation.hpp

```diff
@@ -317,7 +319,554 @@ void mer07_veh02_wang08_nuc_1box(int newnuc_method_user_choice,
+  Real freducea, freduceb;
+  // kk2002 "gamma" parameter (nm2*m2/h)
+  Real gamma_kk;
+  // kk2002 "gr" parameter (nm/h)
+  Real gr_kk;
+  // (kg dry aerosol)/(mol aerosol so4)
+  Real kgaero_per_moleso4a;
+  // "grown" single-particle dry mass (kg)
+  Real mass_part;
+  // (mol aerosol nh4)/(mol aerosol so4)
+  Real molenh4a_per_moleso4a;
+  // actual and bounded nh3 (ppt)
+  // Real nh3ppt;
+  // Real nh3ppt_bb;
+  // kk2002 "nu" parameter (nm)
+  Real nu_kk;
+  // max production of aerosol nh4 over dtnuc (mol/mol-air)
+  Real qmolnh4a_del_max;
+  // max production of aerosol so4 over dtnuc (mol/mol-air)
+  Real qmolso4a_del_max;
+  // nucleation rate (#/m3/s)
+  Real ratenuclt_bb;
+  // nucleation rate after kk2002 adjustment (#/m3/s)
+  Real ratenuclt_kk;
+  // bounded value of rh_in
+  Real rh_bb;
+  // concentration of h2so4 for nucl. calc., molecules cm-3
+  // Real so4vol_in;
+  // bounded value of so4vol_in
+  Real so4vol_bb;
+  // bounded value of temp_in
+  Real temp_bb;
+  // critical-cluster dry volume (m3)
+  Real voldry_clus;
+  // "grown" single-particle dry volume (m3)
+  Real voldry_part;
+  // grown particle (wet-volume)/(dry-volume)
+  Real wetvol_dryvol;
+  // grown particle (dry-volume-from-so4)/(wet-volume)
+  Real wet_volfrac_so4a;
+  // number of h2so4 molecules in the critical nucleus
+  Real cnum_h2so4;
+  // total number of molecules in the critical nucleus
+  Real cnum_tot;
+  // the radius of cluster (nm)
+  Real radius_cluster;
+  // number of nh3 molecules in the critical nucleus
+  Real cnum_nh3;
+  // applied to binary/ternary nucleation rate
+  Real adjust_factor_bin_tern_ratenucl = 1.0;
+  Real adjust_factor_pbl_ratenucl = 1.0;
+
+  // const int icase = 0;
+  // const int icase_reldiffmax = 0;
+  // int lun;
+  int newnuc_method_flagaa2;
+
+  // FIXME: BAD CONSTANTS
+  Real onethird = 1.0 / 3.0;
+  // accomodation coef for h2so4 conden
+  // FIXME: BAD CONSTANT
+  Real accom_coef_h2so4 = 0.65;
+
+  // dry densities (kg/m3) molecular weights of aerosol
+  // ammsulf, ammbisulf, and sulfacid (from mosaic  dens_electrolyte values)
+  //       Real dens_ammsulf   = 1.769e3
+  //       Real dens_ammbisulf = 1.78e3
+  //       Real dens_sulfacid  = 1.841e3
+  // use following to match cam3 modal_aero densities
+  // FIXME: BAD CONSTANTS
+  Real dens_ammsulf = 1770.0;
+  Real dens_ammbisulf = 1770.0;
+  Real dens_sulfacid = 1770.0;
+
+  // molecular weights (g/mol) of aerosol ammsulf, ammbisulf, and sulfacid
+  //    for ammbisulf and sulfacid, use 114 & 96 here rather than 115 & 98
+  //    because we don't keep track of aerosol hion mass
+  // FIXME: BAD CONSTANTS
+  Real mw_ammsulf = 132.0;
+  Real mw_ammbisulf = 114.0;
+  Real mw_sulfacid = 96.0;
+  // Real reldiffmax = 0.0;
+
+  isize_nuc = 1;
+  qnuma_del = 0.0;
+  qso4a_del = 0.0;
+  qnh4a_del = 0.0;
+  qh2so4_del = 0.0;
+  qnh3_del = 0.0;
+  dnclusterdt = 0.0;
+
+  if ((newnuc_method_flagaa != 1) && (newnuc_method_flagaa != 2))
```

Comment:
This is how the box model version had it, and I continued the convention.

Sounds like you'd prefer the +10 version, rather than translating in the interface before calling this?

Reply:
The mam4 refactor repo has the following code:
```
if ((newnuc_method_flagaa /=  1) .and. &
            (newnuc_method_flagaa /=  2) .and. &
            (newnuc_method_flagaa /= 11) .and. &
            (newnuc_method_flagaa /= 12)) return
```

I haven't looked into the code to figure out what all these options are but it will be good to include all these just to be safe.

Reply:
done!